### ISSUE TITLE
truncate_from_right not working on BSD?

### ISSUE THREAD

**Issue by awensaunders** [bhilburn/powerlevel9k#558]
When using the truncate_from_right strategy on a FreeBSD system, sed returns an error and there is no dir string displayed.
`sed: 1: "s@(([^/]{1})([^/]{0}))[ ...": RE error: empty (sub)expression`

zshrc is attached. The problem goes away when using truncate_middle... 

[zshrc.txt](https://github.com/bhilburn/powerlevel9k/files/1111889/zshrc.txt)

**Comment by dritter** [comment]
Thanks for the report! Could you post the output of `sed --version`? That would be helpful.

**Comment by bhilburn** [comment]
Hey @awensaunders - Thanks so much for the bug report! 

Just a ping on @dritter's request so that we can try to debug this `=)`

**Comment by awensaunders** [comment]
Sorry about getting back to you so late. 

`sed --version` isn't even a valid argument on FreeBSD sed (I was surprised by this as well...). I can confirm that it is also a problem in the termux android environment (which uses busybox "not GNU sed version 4"), although it gives the error `bad regex '(([^/]{1})([^/]{0}))[^/]+/': empty (sub)expression`. What's curious is that my mac does not have this same issue, in spite of the fact that it doesn't have GNU sed installed (probably some older BSD derivative, who knows).

**Comment by dritter** [comment]
Interesting. The problem arises from `POWERLEVEL9K_SHORTEN_DELIMITER` being empty. That breaks the regular expression on BSD. I tested this with our BSD VM (FreeBSD 11.0). That seems to be no problem on OSX with an quite old `sed` (from May 10, 2005 as the man page says) installed. The man page on BSD says its from May 10, 2016.

**Comment by awensaunders** [comment]
Yeah, my theory was that it had something to do with the code not checking for BSD itself with sed, but I think perhaps not.

```
# Determine the correct sed parameter.
#
# `sed` is unfortunately not consistent across OSes when it comes to flags.
SED_EXTENDED_REGEX_PARAMETER="-r"
if [[ "$OS" == 'OSX' ]]; then
  local IS_BSD_SED="$(sed --version &>> /dev/null || echo "BSD sed")"
  if [[ -n "$IS_BSD_SED" ]]; then
    SED_EXTENDED_REGEX_PARAMETER="-E"
  fi
fi
```

**Comment by dritter** [comment]
@awensaunders I created #568 for this. Could you check this one out and see if that fixes your problem?

**Comment by bhilburn** [comment]
Fixed by merge of #568!
